src/query/storage/fanout/storage.go

```diff
@@ -219,11 +219,6 @@ func handleFetchResponses(requests []execution.Request) (*storage.FetchResult, e
 			return nil, errors.ErrInvalidFetchResult
 		}
 
-		// NB: if no series to add, continue.
-		if len(fetchreq.result.SeriesList) == 0 {
-			continue
-		}
-
```

Comment:
Ah I see, we still the need the metadata per result now?

Reply:
Maybe mention that in a comment below (we always append at least the metadata from each request, in case of warning headers, etc)?

Reply:
Good call